### ISSUE TITLE
Service Alias Names Containing Periods

### ISSUE THREAD

**Issue by sjiveson** [rancher/rancher#7970]
**Rancher Versions:**
Server: 1.4.1
healthcheck: 0.2.3
ipsec: 0.8.7
network-services: 0.0.14
scheduler: Cattle
kubernetes (if applicable): N/A

**Docker Version:**
 1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**

Boot2Dopcker, Virtualbox guest

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single Node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle

**Steps to Reproduce:**

1) Use a service alias containing periods, for example, www.example.com, using compose v1 syntax (presumably) that worked with Rancher server v1.1.4, with v1.4.1. Example docker-compose:

```
service_name:
  ...
  links:
  - other_service_name:alias.service.name
```

or

2) Use a service alias containing periods using compose v2 syntax with server v1.4.1 (as per: http://docs.rancher.com/rancher/v1.4/en/cattle/adding-service-alias/). Example docker-compose:

```
alias.service.name
  image: rancher/dns-service
  links:
  - other_service_name
```

**Results:** 

1) No error is seen but the alias does not work and lookups against the alias name result in external resolution

2) Rancher compose reports an error that invalid characters are being used in a service name

In summary, we could use www.example.com as a Service Alias prior to v1.2 but due to the (probably more scalable) compose changes for specifying a (now stack wide) Service Alias we no longer can.

**Comment by alena1108** [comment]
@deniseschannon @joshwget  looks like rancher-compose bug. You can create alias with “.” in it via UI/API

**Comment by sjiveson** [comment]
I've finally got around to doing some of my own testing around this. I'm actually not seeing any problem using the GUI or rancher-compose (using the older compose format), with services using managed networking. I'm 99% sure the devs who reported this problem to me (and I sat with for a while for a demo) were using the wrong version of compose. I'm sure they showed me failed attempts using the GUI too, but perhaps that was pasting the new format compose.

**Comment by kaos** [comment]
we see the same issue. We have a service alias, and link to it from another service using an alias for a fqdn. In 1.1.4 the aliased link resolves to the address of the service alias, where as in 1.4.1 it returns the address found on external DNS for the fqdn.

```
  nginx:
    image: xxx:nnn/nginx/master:latest
    external_links:
    - global/api:api.www.7943.xxx.se
    volumes_from:
    - dddd
    labels:
      io.rancher.sidekicks: dddd
      io.rancher.container.hostname_override: container_name
```

in the nginx service, `api.www.7943.xxx.se` does not resolve to the address of `global/api`, as it does in 1.1.4.

**Comment by kaos** [comment]
The rancher dns service has records like this, both on our 1.1.4 and 1.4.1 systems, for the nginx service:

```
  "10.42.248.210": {
    "search": [
      "7921-dddd.rancher.internal",
      "zzz.7921-dddd.rancher.internal",
      "rancher.internal"
    ],
    "recurse": [],
    "authorative": [],
    "a": {
      "api.www.7921.dddd.mr.xxxxx.se.global.rancher.internal.": {
        "answer": [
          "10.42.100.102"
        ]
      },
      "api.www.7921.trinidad.mr.ocean.svenskaspel.se.rancher.internal.": {
        "answer": [
          "10.42.100.102"
        ]
      },
      "qqqq.7921-dddd.rancher.internal.": {
        "answer": [
          "10.42.15.145"
        ]
      },
      "qqqq.rancher.internal.": {
        "answer": [
          "10.42.15.145"
        ]
      }
    },
    "cname": {}
  },
```

When looking up `api.www.7921.dddd.mr.xxxxx.se` on 1.1.4 we get the data from `answers.json` but on 1.4.1 we get the data from our dns servers.

**Comment by joshwget** [comment]
After reading this I think I see three separate issues.

1. Some difference between parsing service names or aliases between v1 and v2 formats. I'm not sure I entirely understand this fully though. @sjiveson can you elaborate on this a bit? Is it the actual service name and not an alias that you're trying to put a period in?

2. Alias lookups don't work with host mode networking.

3. Issue from @kaos which seems more related to the DNS microservice.

Can (2) and (3) be filed as separate issues? It'll be helpful to keep this to just the original issue (1). Thanks!

**Comment by loganhz** [comment]
Close in favor of https://github.com/rancher/rancher/issues/8232
